google/cloud/storage/internal/connection_factory.cc

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/connection_factory.h"
+#include "google/cloud/storage/internal/generic_stub_adapter.h"
+#include "google/cloud/storage/internal/generic_stub_factory.h"
+#include "google/cloud/storage/internal/retry_client.h"
+#include "google/cloud/storage/internal/tracing_client.h"
+#include "google/cloud/internal/opentelemetry.h"
+
+namespace google {
+namespace cloud {
+namespace storage_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+std::shared_ptr<storage::internal::RawClient> DecorateConnection(
+    Options const& opts,
+    std::shared_ptr<storage::internal::RawClient> connection) {
+  auto stub = MakeGenericStubAdapter(std::move(connection));
```

Comment:
`stub = ... connection` 🫨

Reply:
I think we want to get rid of that non-sense, but not in this PR.